### ISSUE TITLE
PowerShell

### ISSUE THREAD

**Issue by JK-Flip-Flop96** [catppuccin/catppuccin#2048]
### Link to repository

https://github.com/JK-Flip-Flop96/powershell

### Screenshots

Flavour summaries:
![image](https://github.com/catppuccin/catppuccin/assets/15033969/4525d269-5547-4994-92b5-fe980317de58)
Getting specific colour values from a palette:
![image](https://github.com/catppuccin/catppuccin/assets/15033969/fd6f4f79-c051-4d8c-9952-4509cdd50d19)

### Any additional comments?

This PowerShell Core Module provides easy access to the Catppuccin Palette for use in PowerShell Scripts. 

I have not provided any screenshots in the README as this module is more of a library, like the Python library. Credit must be given to the Python Library as I based the overall design of this module on it and the colour values were copied directly and converted with Regex to PowerShell syntax.  

The README covers most of the use cases for the module that I've come up with thus far. The examples are mostly taken from my [personal PowerShell Profile](https://github.com/JK-Flip-Flop96/powershell-profile/blob/master/Microsoft.PowerShell_profile.ps1) except for the Prompt as my whole Prompt function would be over the top for a demo. 

Ideally, it'd be good to make this available via the PowerShell Gallery but I have not explored that at this time. 

Apologies in advance for the inconsistent use of British vs American English. I'm a British English user myself but couldn't keep either consistent throughout. 😆

### Submission Guidelines

- [X] I have read and followed Catppuccin's submission guidelines.

**Comment by nyxkrage** [comment]
Hey, just a couple of things! Most of the code looks really good.

1. We should memoize the HSL values rather than needing to generate them all the time, for example by adding 4 new properties, `[bool]$HasHSL`, `[double]$Hue`, `[double]$Saturation`, and `[double]$Lightness`, and then assigning to them on the first time you run the Hsl function otherwise if $HasHSL is already set, just return the respective variables, since I feel that if you need the HSL variables, you're probably gonna be using them multiple times.

2. Please add the copyright notice from the license into the manifest Copyright property.

3. If you could populate PSData with LicenseUri and ProjectUri pointing to https://github.com/catppuccin/powershell/blob/main/LICENSE and https://github.com/catppuccin/powershell respectively

4. I think it would make sense to at least change the "Catppuccin for PowerShell" header to specifically say PowerShell 7 or PowerShell Core, at least until it is also compatible with the older PowerShell 5 that Windows ships with

After that this should be ready to transfer!

**Comment by JK-Flip-Flop96** [comment]
Great, I'll get to work on these changes soon. I think I will go ahead and work on Powershell 5.1 support as it should really just be a case of removing all of the instances of "`e" from the code. 

Thanks for providing the correct info for the PSData and Copyright info.

**Comment by nyxkrage** [comment]
Yeah, this should be good to go now, letting @nekowinston take over from here, as he can add you to the organization and make sure it gets transferred over properly!

**Comment by nekowinston** [comment]
Thanks for the ping. @JK-Flip-Flop96 I've sent you an invite, feel free to transfer the repository :rocket:
